crates/crypto/proof-params/src/lib.rs

```diff
@@ -17,12 +19,52 @@ pub use traits::{
 #[cfg(feature = "proving-keys")]
 mod proving_keys;
 
-/// Proving key for the spend proof.
 #[cfg(feature = "proving-keys")]
-pub static SPEND_PROOF_PROVING_KEY: Lazy<ProvingKey<Bls12_377>> =
-    Lazy::new(proving_keys::spend_proving_parameters);
+#[derive(Debug, Default)]
+pub struct LazyProvingKey {
+    inner: OnceCell<ProvingKey<Bls12_377>>,
+}
+
+#[cfg(feature = "proving-keys")]
+impl LazyProvingKey {
+    // Lazily construct proving key
+    pub fn new() -> Self {
+        LazyProvingKey { 
+            inner: OnceCell::new(),
+        }
+    }
+
+    pub fn load(&self, key: &str) -> ProvingKey<Bls12_377> {
```

Comment:
I think `load` should take a byte slice, and then attempt to deserialize into a ProvingKey in this method

Reply:
Yes I haven't changed the type signature yet!

Reply:
The type signature for the load method has been changed accordingly.